# Generated by Django 2.0.2 on 2018-08-30 17:25

from django.db import migrations


def migrate_openstax_books(apps, schema_editor):
    Errata = apps.get_model('errata', 'Errata')
    for errata in Errata.objects.all():
        errata.openstax_book = errata.book.title
        errata.save()

def migrate_user_accounts(apps, schema_editor):
    Errata = apps.get_model('errata', 'Errata')
    UserSocialAuth = apps.get_model('social_django', 'UserSocialAuth')
    for errata in Errata.objects.all():
        try:
            user = UserSocialAuth.objects.filter(user=errata.submitted_by)[0]
            errata.submitted_by_account_id = user.uid
            errata.save()
        except IndexError:
            print("Problem with Errata ID (NO USER): {}".format(errata.id))

class Migration(migrations.Migration):

    dependencies = [
        ('errata', '0028_auto_20180830_1220'),
    ]

    operations = [
        migrations.RunPython(migrate_openstax_books),
        migrations.RunPython(migrate_user_accounts),
    ]
